Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -5071,28 +5071,22 @@ void CPhysicalGeometry::ComputeNSpan(CConfig *config, unsigned short val_iZone,
   }
   else{
     if(SpanWise_Kind == AUTOMATIC){
-      /*--- loop to find inflow of outflow marker---*/
+      /*--- loop to find inflow or outflow marker---*/
       for (iMarker = 0; iMarker < nMarker; iMarker++){
         for (iMarkerTP=1; iMarkerTP < config->GetnMarker_Turbomachinery()+1; iMarkerTP++){
-
           if (config->GetMarker_All_Turbomachinery(iMarker) != iMarkerTP) continue;
-          if (config->GetMarker_All_TurbomachineryFlag(iMarker) != marker_flag) continue;
+          if (config->GetMarker_All_TurbomachineryFlag(iMarker) != marker_flag) continue; 
+          for (iVertex = 0; iVertex < nVertex[iMarker]; iVertex++) {
+            iPoint = vertex[iMarker][iVertex]->GetNode();
 
-          /*--- loop to find the vertex that ar both of inflow or outflow marker and on the periodic
-           * in order to caount the number of Span ---*/
-          for (jMarker = 0; jMarker < nMarker; jMarker++){
+            /*--- loop to find the vertex that ar both of inflow or outflow marker and on the periodic
+              * in order to count the number of Span ---*/
+            for (jMarker = 0; jMarker < nMarker; jMarker++){
             if (config->GetMarker_All_KindBC(jMarker) != PERIODIC_BOUNDARY) continue;
-
-            for (iVertex = 0; iVertex < nVertex[iMarker]; iVertex++) {
-
-              iPoint = vertex[iMarker][iVertex]->GetNode();
-              if (!nodes->GetDomain(iPoint)) continue;
-
               PeriodicBoundary = config->GetMarker_All_PerBound(jMarker);
               jVertex = nodes->GetVertex(iPoint, jMarker);
-
-              if ((jVertex != -1) && (PeriodicBoundary == (val_iZone + 1))){
-                nSpan++;
+              if ((jVertex != -1) && (PeriodicBoundary == ((val_iZone) + 1)) && nodes->GetDomain(iPoint)) {
+                  nSpan++;
```

Comment:
Was my fix of this function incorrect?

Reply:
There were a lot of differences in this file, so I tried to stick to the algorithm the was working than to have more uncertainty.

Reply:
I could argue the same, since I went out of my way to debug this problem for this user #1219 when no one else did. So, if it is all the same, put it back.